### PR TITLE
[GJ-90] Support ClickHouse DataSource V2 -- Phase one

### LINKED ISSUE #90: Support ClickHouse Spark DataSource V2 -- Phase One
Support to use ClickHouse Spark DataSource V2 to create ClickHouse tables, refresh tables, and select tables (TPCH-Q6).

### PR DESCRIPTION
**Pull request by zzcclp**
Support to use ClickHouse Spark DataSource V2 to : 
1. create ClickHouse tables;
2. refresh tables;
3. query tables (TPCH-Q6);

To enabled this feature, add this conf in spark:
spark.sql.catalog.spark_catalog=org.apache.spark.sql.execution.datasources.v2.clickhouse.ClickHouseSparkCatalog

Write ClickHouse MergeTree data with Spark DataSource V2 will be implemented on Phase two.

To enable native engine, below option should be added:
--conf spark.oap.sql.enable.native.engine=true

Closes #90

### CODE REVIEW

File: jvm/src/main/scala/com/intel/oap/execution/WholeStageTransformerExec.scala [hunk 1]
```diff
@@ -313,32 +308,33 @@ case class WholeStageTransformerExec(child: SparkPlan)(val transformStageId: Int
 
       val startTime = System.nanoTime()
       val substraitPlanPartition = fileScan.getPartitions.map {
+        case p: NativeMergeTreePartition => {
```
Comment (rui-mo) [inline jvm/src/main/scala/com/intel/oap/execution/WholeStageTransformerExec.scala:311]: format: no need to add "{" here.

File: jvm/src/main/scala/com/intel/oap/execution/WholeStageTransformerExec.scala [hunk 2]
```diff
@@ -313,32 +308,33 @@ case class WholeStageTransformerExec(child: SparkPlan)(val transformStageId: Int
...
+          val extensionTableNode = ExtensionTableBuilder.makeExtensionTable(p.minParts, p.maxParts,
```
Comment (rui-mo) [inline jvm/src/main/scala/com/intel/oap/execution/WholeStageTransformerExec.scala:312]: format: make line shorter.

File: jvm/src/main/scala/com/intel/oap/execution/WholeStageTransformerExec.scala [hunk 3]
```diff
@@ -313,32 +308,33 @@ case class WholeStageTransformerExec(child: SparkPlan)(val transformStageId: Int
...
+            p.database, p.table, p.tablePath)
+          wsCxt.substraitContext.setExtensionTableNode(extensionTableNode)
+          //logWarning(s"The substrait plan for partition " +
```
Comment (rui-mo) [inline jvm/src/main/scala/com/intel/oap/execution/WholeStageTransformerExec.scala:315]: format: add space after "//".

File: jvm/src/main/scala/com/intel/oap/extension/ColumnarOverrides.scala [hunk 4]
```diff
@@ -316,7 +316,9 @@ case class ColumnarOverrideRules(session: SparkSession) extends ColumnarRule wit
     plan.logicalLink.isDefined
 
   override def preColumnarTransitions: Rule[SparkPlan] = plan => {
-    if (columnarEnabled) {
+    if (nativeEngineEnabled && !plan.isInstanceOf[SerializeFromObjectExec] &&
+      !plan.isInstanceOf[ObjectHashAggregateExec] &&
+      !plan.isInstanceOf[V2CommandExec]) {
```
Comment (rui-mo) [inline jvm/src/main/scala/com/intel/oap/extension/ColumnarOverrides.scala:321]: Could you add some explanations on why need to exclude ObjectHashAggregateExec and V2CommandExec here?

File: jvm/src/main/scala/com/intel/oap/extension/ColumnarOverrides.scala [hunk 5]
```diff
@@ -327,7 +329,9 @@ case class ColumnarOverrideRules(session: SparkSession) extends ColumnarRule wit
   }
 
   override def postColumnarTransitions: Rule[SparkPlan] = plan => {
-    if (columnarEnabled) {
+    if (nativeEngineEnabled && !plan.isInstanceOf[SerializeFromObjectExec] &&
+      !plan.isInstanceOf[ObjectHashAggregateExec] &&
+      !plan.isInstanceOf[V2CommandExec]) {
```
Comment (rui-mo) [inline jvm/src/main/scala/com/intel/oap/extension/ColumnarOverrides.scala:334]: Same as above.

**Comment by rui-mo** [comment]
Verified locally.

**Comment by rui-mo** [comment]
@marin-ma do you have any comment?

File: jvm/src/main/scala/com/intel/oap/execution/WholeStageTransformerExec.scala (hunk 1 above)
Comment (zzcclp) [reply jvm/src/main/scala/com/intel/oap/execution/WholeStageTransformerExec.scala]: Done

File: jvm/src/main/scala/com/intel/oap/execution/WholeStageTransformerExec.scala (hunk 2 above)
Comment (zzcclp) [reply jvm/src/main/scala/com/intel/oap/execution/WholeStageTransformerExec.scala]: Done

File: jvm/src/main/scala/com/intel/oap/execution/WholeStageTransformerExec.scala (hunk 3 above)
Comment (zzcclp) [reply jvm/src/main/scala/com/intel/oap/execution/WholeStageTransformerExec.scala]: Done

File: jvm/src/main/scala/com/intel/oap/extension/ColumnarOverrides.scala (hunk 4 above)
Comment (zzcclp) [reply jvm/src/main/scala/com/intel/oap/extension/ColumnarOverrides.scala:328]: Added some comments, and create a issue to fix fallback issues: #95

File: jvm/src/main/scala/com/intel/oap/extension/ColumnarOverrides.scala (hunk 5 above)
Comment (zzcclp) [reply jvm/src/main/scala/com/intel/oap/extension/ColumnarOverrides.scala:341]: Done

**Comment by marin-ma** [comment]
LGTM

**Pull request merged**